### PR TITLE
made copy materialized data view distinct

### PR DESCRIPTION
**Pull request by MartinHinz**
**Problem:**

The current CSV export generates significantly more rows than the number of data sets, due to the many-to-many relationships with references, typo-eco-units, and periods. This leads to unexpected results for users who rely on a clean, 1:1 representation of the data.

**Solution:**

This PR implements a quick fix to eliminate the extra rows during the export process. While it addresses the immediate issue, we recommend addressing the root cause by refining the creation logic of the data_views materialized view to filter out these additional rows at the source.

**Value:**

Even if the materialized view is updated in the future, this PR adds an additional safeguard to ensure exported data meets user expectations.

### CODE REVIEW

File: app/controllers/data_controller.rb [hunk 1]
```diff
@@ -37,18 +37,28 @@ def index
 ) AS geojson from (" + @sites.to_sql+ ") AS subquery1) AS subquery2").to_sql)[0]['measurements'], adapter: nil, serializer: nil        
       }
       format.csv {
-        query = "COPY (SELECT * FROM data_views WHERE id IN (" + @data.xrons.pluck(:id).join(', ').to_s + ") ) TO STDOUT WITH CSV HEADER"
-        
+        query = <<~SQL
+          COPY (
+            SELECT DISTINCT "id", "labnr", "bp", "std", "cal_bp", "cal_std", "delta_c13", "source_database", "lab_name", "material", "species", "feature", "feature_type", "site", "country", "lat", "lng", "site_type",
+                   periods::text AS periods,
+                   typochronological_units::text AS typochronological_units,
```
Comment (joeroe) [inline app/controllers/data_controller.rb:44]: What does this refer to? There's no such table (or shouldn't be).

File: app/controllers/data_controller.rb [hunk 2]
```diff
@@ -37,18 +37,28 @@ def index
...
+                   ecochronological_units::text AS ecochronological_units,
```
Comment (joeroe) [inline app/controllers/data_controller.rb:45]: What does this refer to? There's no such table (or shouldn't be).

File: app/controllers/data_controller.rb (hunk 1 above)
Comment (MartinHinz) [reply app/controllers/data_controller.rb:44]: This comes in the materialized view from typos, stays there and in the csv for legacy reasons. Problem?

File: app/controllers/data_controller.rb (hunk 2 above)
Comment (MartinHinz) [reply app/controllers/data_controller.rb:45]: This comes in the materialized view from typos, stays there and in the csv for legacy reasons. Problem?

Comment (joeroe) [reply app/controllers/data_controller.rb:45]: Ah right, didn't realise there were multiple mat views.

**Review by joeroe — APPROVED** [review]

**Pull request closed without merging**